### PR TITLE
Made wilson uploader limits configurable

### LINKED ISSUE #2006: Make wilson uploader configurable from cluster config

### PR DESCRIPTION
**Pull request by domwst**
### Made wilson uploader limits configurable

...

### Changelog category 

* Not for changelog (changelog entry is not required)

### Additional information

...

### CODE REVIEW

File: ydb/library/actors/wilson/wilson_uploader.h [hunk 1]
```diff
@@ -30,6 +30,10 @@ namespace NWilson {
         TString ServiceName;
         std::unique_ptr<IGrpcSigner> GrpcSigner;
 
+        ui32 MaxSpansPerSecond = 10;
```
Comment (domwst) [inline ydb/library/actors/wilson/wilson_uploader.h:33]: Not sure whether these limits should be in this file or in kikimr_services_initializers.cpp

Comment (alexvru) [reply ydb/library/actors/wilson/wilson_uploader.h:33]: I'd suggest no default limits when no respective configuration section is filled in, especially in OpenSource.

Comment (domwst) [reply ydb/library/actors/wilson/wilson_uploader.h]: Changed default limits to infinity effectively

File: ydb/core/protos/config.proto [hunk 2]
```diff
@@ -1549,21 +1549,30 @@ message TTracingConfig {
...
+        optional uint64 MaxRatePerMinute = 4;
+        optional uint64 MaxBurst = 5;
     }
 
     message TExternalThrottlingScope {
         optional TSelectors Scope = 1;
-        optional uint32 MaxRatePerMinute = 2;
-        optional uint32 MaxBurst = 3;
+        optional uint64 MaxRatePerMinute = 2;
+        optional uint64 MaxBurst = 3;
```
Comment (alexvru) [inline ydb/core/protos/config.proto:1559]: Some description would be great. For other fields too. Purpose, units.

File: ydb/library/actors/wilson/wilson_uploader.cpp [hunk 3]
```diff
@@ -136,28 +141,37 @@ namespace NWilson {
...
+                    ALOG_WARN(WILSON_SERVICE_ID, "max_spans_in_batch shold be greater than 0, changing to 1");
+                    MaxSpansInBatch = 1;
+                }
+
                 TStringBuf scheme;
                 TStringBuf host;
                 ui16 port;
                 if (!TryGetSchemeHostAndPort(CollectorUrl, scheme, host, port)) {
-                    LOG_ERROR_S(*TlsActivationContext, WILSON_SERVICE_ID, "Failed to parse collector url (" << CollectorUrl << " was provided). Wilson wouldn't work");
+                    ALOG_ERROR(WILSON_SERVICE_ID, "Failed to parse collector url (" << CollectorUrl << " was provided). Wilson wouldn't work");
```
Comment (alexvru) [inline ydb/library/actors/wilson/wilson_uploader.cpp:157]: We need some indicator here, health check light or a sensor. Do we have one?

File: ydb/core/protos/config.proto [hunk 4]
```diff
@@ -1549,21 +1549,30 @@ message TTracingConfig {
...
+    }
+
+    message TUploaderConfig {
+        optional uint64 MaxSpansPerSecond = 1;
```
Comment (alexvru) [inline ydb/core/protos/config.proto:1563]: Spans per second, but rate per minute? I'd prefer common units here.

File: ydb/core/protos/config.proto [hunk 5]
```diff
@@ -1549,21 +1549,30 @@ message TTracingConfig {
...
+        optional uint64 MaxSpansInBatch = 2;
+        optional uint64 MaxBytesInBatch = 3;
+        optional uint64 MaxBatchAccumulationMilliseconds = 4;
+        optional uint32 SpanTtlSeconds = 5;
```
Comment (alexvru) [inline ydb/core/protos/config.proto:1567]: What is "span TTL"? I'd expect time the span is alive in Jaeger, after which it is deleted. Here it actually means approximate time after span is deleted unless sent to collector.

File: ydb/library/actors/wilson/wilson_uploader.cpp [hunk 6]
```diff
@@ -136,28 +141,37 @@ namespace NWilson {
...
+                    ALOG_ERROR(WILSON_SERVICE_ID, "Wrong scheme provided: " << scheme << " (only grpc:// and grpcs:// are supported). Wilson wouldn't work");
                     Become(&TThis::StateBroken);
                     return;
                 }
                 Channel = grpc::CreateChannel(TStringBuilder() << host << ":" << port,
                                               scheme == "grpcs://" ? grpc::SslCredentials({}) : grpc::InsecureChannelCredentials());
                 Stub = NServiceProto::TraceService::NewStub(Channel);
 
-                LOG_INFO_S(*TlsActivationContext, WILSON_SERVICE_ID, "TWilsonUploader::Bootstrap");
+                ALOG_INFO(WILSON_SERVICE_ID, "TWilsonUploader::Bootstrap");
```
Comment (alexvru) [inline ydb/library/actors/wilson/wilson_uploader.cpp:169]: If we have an indicator of working/erroneous Wilson, we can reduce level to DEBUG here.

**Review by alexvru — CHANGES_REQUESTED** [review]
Consider making a light or sensor indicating:
1. Wilson is working.
2. Wilson is in error state.
3. Wilson is disabled.

File: ydb/library/actors/wilson/wilson_uploader.cpp (hunk 3 above)
Comment (domwst) [reply ydb/library/actors/wilson/wilson_uploader.cpp:157]: We don't at the moment, let's do it in a separate PR

File: ydb/library/actors/wilson/wilson_uploader.cpp (hunk 6 above)
Comment (domwst) [reply ydb/library/actors/wilson/wilson_uploader.cpp:169]: Do we have one?

File: ydb/core/protos/config.proto (hunk 4 above)
Comment (domwst) [reply ydb/core/protos/config.proto:1563]: But they limit different things, which often have different scale. Per-minute limits control traces, that often have quite some spans. But for consistency we centanly can  limit upload rate per minute as well

**Review by alexvru — APPROVED** [review]

**Pull request merged**